components/scream/scripts/gather-all-data

```diff
@@ -70,14 +90,36 @@ OR
     expect(not (args.local and len(args.machines) > 1),
            "Cannot run on multiple machines if local")
 
-    if args.local:
-        expect(os.path.isdir("scream"), "Expected to see ./scream")
+    # Compute root!
+    script_home = pathlib.Path(__file__).resolve().parent.parent
+    if not args.root_dir:
+        if args.local:
+            if args.scream_docs:
+                args.root_dir = pathlib.Path().resolve() # cwd
+                expect(args.root_dir.name == "micro-apps",
+                       "Please do local scream-docs gathers from $scream-docs-repo/micro-apps")
+            else:
+                args.root_dir = script_home
+        else:
+            if args.scream_docs:
+                args.root_dir = pathlib.Path("~/scream-docs-perf-$machine/micro-apps")
+            else:
+                args.root_dir = pathlib.Path("~/scream-perf-$machine/components/scream")
+
+    else:
+        if args.local:
+            args.root_dir = pathlib.Path(args.root_dir).resolve()
+        else:
+            args.root_dir = pathlib.Path(args.root_dir)
+
+    # If doing remote testing, set remote repos to commit matching local commit
+    if not args.commit and not args.local:
         if args.scream_docs:
-            expect(os.path.isdir("scream-docs"), "Expected to see ./scream-docs")
-
-    if not args.commit:
-        expect(args.local, "Could not determine commit")
-        args.commit = get_current_commit(short=True, repo="./scream") if not args.scream_docs else get_current_commit(short=True, repo="./scream-docs")
+            args.commit = get_current_commit(short=True)
+            expect(args.commit, "Must be able to problem scream-docs commit. "
```

Comment:
Maybe 'problem' should be 'probe'. Not sure I understand the sentence otherwise,

Reply:
Good catch.